### PR TITLE
Fix the reported Compound supply values – LL-5470

### PR DESCRIPTION
**Pull request by stephenjust**
Previously, the reported supply balance was inaccurate. First, we were
using the transaction time to look up the historical exchange rate. If
the tx was in the mempool for a while, this could create a significant
delta to when the block was written.

* Updated fetchHistoricalRates to use the block depth, this is more
  accurate than querying based on transaction time.
* Use the underlying token transaction instead of using the historical
  rate for calculation. Even the block number-based implementation has
  noticeable error.

Technically, fetchHistoricalRates isn't needed anymore, but I have left it in
place to not affect anything that may depend on `extra.rate`.

### CODE REVIEW

**Comment by gre** [comment]
@stephenjust thanks for your contribution! we'll review the PR in our sprint.

**Comment by gre** [comment]
@stephenjust can you run `yarn prettier` ? thanks

File: src/families/ethereum/modules/compound.js [hunk 1]
```diff
@@ -396,11 +396,16 @@ export function prepareTokenAccounts(
   return subAccounts.concat(implicitCTokenAccounts);
 }
 
-const cdaiToDaiOpMapping: { [_: OperationType]: ?OperationType } = {
+const ctokenToGeneratedTokenOpMapping: { [_: OperationType]: ?OperationType } = {
```
Comment (gre) [inline src/families/ethereum/modules/compound.js:399]: @stephenjust can you run `yarn prettier` ? thanks

**Review by gre — CHANGES_REQUESTED** [review]

**Review by JunichiSugiura — APPROVED** [review]

File: src/families/ethereum/modules/compound.js (hunk 1 above)
Comment (stephenjust) [reply src/families/ethereum/modules/compound.js]: Looks like Junichi got it. Thanks!

File: src/families/ethereum/modules/compound.js [hunk 2]
```diff
@@ -471,23 +478,35 @@ export async function digestTokenAccounts(
         // cOUT => REDEEM
         const rates = await fetchHistoricalRates(
           ctoken,
-          ctokenAccount.operations.map((op) => op.date)
+          ctokenAccount.operations.reduce(
+            (prev, op) => (op.blockHeight ? [...prev, op.blockHeight] : prev),
```
Comment (gre) [inline src/families/ethereum/modules/compound.js:482]: - maybe reduce+spread is not helping perf, could do more specific logic maybe?
- would we want to dedup even tho there are low chance to have two ops on same block?

**Review by gre — COMMENTED** [review]
overall it's ok. i have a small comment on one part.

I think we'll wait to validate internally the fix and we will be good to merge.

**Comment by gre** [comment]
I've tested and compared on my account the two implementations, on the small amounts I have on my test account there are no significant change only 1 cent here and here, which means it's a GO for this PR as it seems to slightly adjust the number to their more correct block.

Thanks @stephenjust for the contribution.

**Pull request merged**